### PR TITLE
samples: nrf_desktop: Battery characteristic

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds non-linear battery characteristic.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@pdunaj, @Qbicz, could yout review ?

File: samples/nrf_desktop/configuration/nrf52840_pca20041/battery_lut.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* Converting measured battery voltage[mV] to State of Charge[%].
```
Comment (pdunaj) [inline samples/nrf_desktop/configuration/nrf52840_pca20041/battery_lut.h]: Add copyright header and multi-include protection (just like in other xyz_def files)

Comment (MarekPieta) [reply samples/nrf_desktop/configuration/nrf52840_pca20041/battery_lut.h]: done

Comment (MarekPieta) [reply samples/nrf_desktop/configuration/nrf52840_pca20041/battery_lut.h]: done

File: samples/nrf_desktop/src/hw_interface/Kconfig [hunk 2]
```diff
@@ -194,6 +194,14 @@ config DESKTOP_BATTERY_VOLTAGE_DIVIDER_UPPER
 	help
 	  Output voltage is measured on the lower resistor.
 
+config DESKTOP_VOLTAGE_TO_SOC_DELTA
+	int "Difference between adjanced elements in conversion LUT [mV]"
```
Comment (Qbicz) [inline samples/nrf_desktop/src/hw_interface/Kconfig]: adjacent

File: samples/nrf_desktop/src/hw_interface/battery_discrete.c [hunk 3]
```diff
@@ -146,6 +147,13 @@ static int init_adc(void)
 		return err;
 	}
 
+	/* Check if number of elements in LUT is proper */
+	static_assert(CONFIG_DESKTOP_BATTERY_MAX_LEVEL
+		      - CONFIG_DESKTOP_BATTERY_MIN_LEVEL
+		      == (ARRAY_SIZE(battery_voltage_to_soc) - 1)
+		      * CONFIG_DESKTOP_VOLTAGE_TO_SOC_DELTA,
+		      "Inproper number of elements in lookup table");
```
Comment (Qbicz) [inline samples/nrf_desktop/src/hw_interface/battery_discrete.c]: Improper number (or "Wrong number")

File: samples/nrf_desktop/configuration/nrf52840_pca20041/battery_def.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * Violating this requirement triggers a multiple definition error at link time.
+ */
+const struct {} battery_def_include_once;
+
+/* Converting measured battery voltage[mV] to State of Charge[%].
+ * First element corresponds to CONFIG_DESKTOP_BATTERY_MIN_LEVEL.
+ * Each element is CONFIG_DESKTOP_VOLTAGE_TO_SOC_DELTA higher than previous.
+ * Defined separately for every configuration.
+ */
+static const u8_t battery_voltage_to_soc[] = {
```
Comment (Qbicz) [inline samples/nrf_desktop/configuration/nrf52840_pca20041/battery_def.h]: Is it the characteristic for Thingy battery or was it measured for the 3-pin battery that we now have?

File: samples/nrf_desktop/src/hw_interface/Kconfig (hunk 2 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/hw_interface/Kconfig]: done

File: samples/nrf_desktop/src/hw_interface/battery_discrete.c (hunk 3 above)
Comment (MarekPieta) [reply samples/nrf_desktop/src/hw_interface/battery_discrete.c]: ok, changed to Improper

File: samples/nrf_desktop/configuration/nrf52840_pca20041/battery_def.h (hunk 4 above)
Comment (MarekPieta) [reply samples/nrf_desktop/configuration/nrf52840_pca20041/battery_def.h]: It was from Thingy battery, we may changed it later on to match hardware configuration

**Pull request merged**